fdbclient/include/fdbclient/MetaclusterManagement.actor.h

```diff
@@ -1544,34 +1544,83 @@ Future<Void> deleteTenant(Reference<DB> db, int64_t id) {
+                                                                           int limit,
+                                                                           int offset = 0) {
+	auto future = ManagementClusterMetadata::tenantMetadata().tenantNameIndex.getRange(tr, begin, end, limit + offset);
+	return fmap(
+	    [offset](auto f) {
+		    std::vector<std::pair<TenantName, int64_t>>& results = f.results;
+		    results.erase(results.begin(), results.begin() + offset);
+		    return results;
+	    },
+	    future);
+}
 
-	state KeyBackedRangeResult<std::pair<TenantName, int64_t>> matchingTenants =
-	    wait(ManagementClusterMetadata::tenantMetadata().tenantNameIndex.getRange(tr, begin, end, limit));
+template <class DB>
+Future<std::vector<std::pair<TenantName, int64_t>>> listTenants(Reference<DB> db,
+                                                                TenantName begin,
+                                                                TenantName end,
+                                                                int limit,
+                                                                int offset = 0) {
+	return runTransaction(db, [=](Reference<typename DB::TransactionT> tr) {
+		tr->setOption(FDBTransactionOptions::LOCK_AWARE);
+		tr->setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
+		return listTenantsTransaction(tr, begin, end, limit, offset);
+	});
+}
 
+ACTOR template <class Transaction>
+Future<std::vector<std::pair<TenantName, TenantMapEntry>>> listTenantMetadataTransaction(Transaction tr,
+                                                                                         TenantNameRef begin,
+                                                                                         TenantNameRef end,
+                                                                                         int limit) {
+	state std::vector<std::pair<TenantName, int64_t>> matchingTenants =
+	    wait(listTenantsTransaction(tr, begin, end, limit));
 	state std::vector<Future<TenantMapEntry>> tenantEntryFutures;
-	for (auto const& [name, id] : matchingTenants.results) {
+	for (auto const& [name, id] : matchingTenants) {
 		tenantEntryFutures.push_back(getTenantTransaction(tr, id));
 	}
 
 	wait(waitForAll(tenantEntryFutures));
 
 	std::vector<std::pair<TenantName, TenantMapEntry>> results;
-	for (int i = 0; i < matchingTenants.results.size(); ++i) {
+	for (int i = 0; i < matchingTenants.size(); ++i) {
 		// Tenants being renamed will show up twice; once under each name
-		results.emplace_back(matchingTenants.results[i].first, tenantEntryFutures[i].get());
+		results.emplace_back(matchingTenants[i].first, tenantEntryFutures[i].get());
+	}
+
+	return results;
+}
+
+// read tenant entries of tenantIds
+ACTOR template <class Transaction>
+Future<std::vector<std::pair<TenantName, TenantMapEntry>>> listTenantMetadataTransaction(
+    Transaction tr,
+    std::vector<std::pair<TenantName, int64_t>> tenantIds) {
+
+	state int i = 0;
```

Comment:
I'd recommend not using a name like `i` for a state variable, as it can lead to weird issues with variable shadowing. For this, you could use something like `tenantIdIndex`.